### DISCUSSION TITLE
Synchronous or Asynchronous

### DISCUSSION THREAD

**Discussion by Nega100** [jspsych/jsPsych#2282] [Q&A] [answered]
Hi Josh, Becky and all JsPsych User
I have two question :
1. Is JsPsych synchronous or asynchronous?
2. Can I add like MelonJS for gaming in JsPsych Framework?if possible could this affect the rt's or loading time maybe. advantages/disadvantages.

Thanks for the time and answer. Highly appreciated
Nega

**Comment by jodeleeuw** [comment, marked as answer]
Hi @Nega100 

> Is JsPsych synchronous or asynchronous?

It has both synchronous and asynchronous components. Is there something in particular you are thinking about? I think some more context for the question would help me provide a clearer answer.

> Can I add like MelonJS for gaming in JsPsych Framework?

You can add other libraries. A typical way to do this would be to use the library in a custom plugin. I know some folks who have successfully merged [Phaser](https://phaser.io/) with jsPsych plugins. I'm not sure if MelonJS specifically will work, but my quick glance at their docs suggest that it probably would.
